pkg/model/firewall.go

```diff
@@ -36,12 +39,25 @@ const (
 // FirewallModelBuilder configures firewall network objects
 type FirewallModelBuilder struct {
 	*KopsModelContext
+	Cloud     awsup.AWSCloud
 	Lifecycle *fi.Lifecycle
 }
 
 var _ fi.ModelBuilder = &FirewallModelBuilder{}
 
 func (b *FirewallModelBuilder) Build(c *fi.ModelBuilderContext) error {
+
+	tasks, err := b.getExistingRulesFromCloud()
+	if err != nil {
+		return err
+	}
+
+	finalRules := make(map[string]*awstasks.SecurityGroupRule)
```

Comment:
```suggestion
	finalRules := make(map[string]*awstasks.SecurityGroupRule, len(tasks))
```
Though I suppose that should really be the maximum of `len(tasks)` and `len(b.SecurityGroupRules)`.